chain/network/src/peer_manager.rs

```diff
@@ -357,6 +358,23 @@ impl PeerManagerActor {
             warn!(target: "network", "Unknown account {} in peers, not supported indirect routing", account_id);
         }
     }
+
+    fn process_announce_account(&mut self, peer_id: PeerId, account_id: AccountId, signature: Signature) -> AnnounceAccountResult {
+        if self.account_peers.contains_key(&account_id) {
+            // This announcement was already received.
+            return AnnounceAccountResult::IgnoreAnnounce;
+        }
+
+        // Verify the signature.
+        if !verify(account_id.as_bytes(), &signature, &peer_id.pub_key()) {
+            return AnnounceAccountResult::InvalidSignature;
+        }
+
+        // TODO: Establish direct connection with peer if not in `self.active_peers`
```

Comment:
validators may have way too many connections. so not always establishing direct connection is the best thing - instead keep track a "route" - who is the closest peer who already has connected to this validator.
This way, when we need to send something to given account_Id, we will look up this "relay" and use it.
Also will require new message type "Relay".